Automattic/wp-calypso
# Jetpack Backup: Refactor and unit test the Backup page date picker

#### Changes proposed in this Pull Request

* Combine `client/components/jetpack/backup-date-picker` with `client/my-sites/backup/date-picker`, since the latter is only a wrapper for the former, and they're only used in the context of the Backup page.
* Convert the new `BackupDatePicker` from a class-based component to a functional one that takes advantage of React hooks.
* Convert the new `BackupDatePicker` from JavaScript to TypeScript.
* Add unit tests to cover `BackupDatePicker` and its hooks.

#### Testing instructions (regression)

On both Calypso Blue and Green:

1. Navigate to the Backup page.
2. Verify that you can navigate forward and backward, but only within the range of today (at the latest) to the oldest known backup attempt (at the earliest). _(For sites with Backup Daily, the earliest available date may likely be 30 days in the past.)_
3. Select a site whose timezone has an offset more negative than your browser's (e.g., if your browser uses UTC-6, set your site to use UTC-8). Verify you can navigate both forward and backward as normal, like in step 2. _(See #48269 for more details on this.)_
4. Manually alter your browser's URL to select a date that's sometime in the future. Verify that you can navigate backward toward the present, but not any further forward in time.
5. Do the same as in step 3, but select a date that's prior to the earliest known backup attempt. Verify that you can navigate forward, but not any further backward in time.
